### PR TITLE
JIT: fix synthetic profile check option, track improper headers

### PR DESCRIPTION
**Pull request by AndyAyersMS**
Fix the config option that enables consistency checking for profile synthesis.

Make the loop package improper header information visible to synthesis, so it can avoid checking cases that will be inconsistent.

Contributes to #93020.

### CODE REVIEW

**Comment by AndyAyersMS** [comment]
@jakobbotsch PTAL @amanasifkhalid FYI
cc @dotnet/jit-contrib

**Comment by AndyAyersMS** [comment]
CI is actually done, despite status above. Just one failing test. Profile synthesis happens before importation and so we may run into bad IL&mdash;cases where there is no sensical profile. Not quite sure what to do about it just yet as my original goal was to be able to say that the importer would always see consistent profile data.

I suppose can just have a deferred assertions -- if importation succeeds, we can assert that before importation, the profile was consistent.

File: src/coreclr/jit/fgprofilesynthesis.cpp [hunk 1]
```diff
@@ -31,8 +31,9 @@
 //
 void ProfileSynthesis::Run(ProfileSynthesisOption option)
 {
-    m_dfsTree = m_comp->fgComputeDfs();
-    m_loops   = FlowGraphNaturalLoops::Find(m_dfsTree);
+    m_dfsTree             = m_comp->fgComputeDfs();
+    m_loops               = FlowGraphNaturalLoops::Find(m_dfsTree);
+    m_improperLoopHeaders = m_loops->ImproperLoopHeaders();
```
Comment (jakobbotsch) [inline src/coreclr/jit/fgprofilesynthesis.cpp:36]: Nit: remove the field from `ProfileSynthesis` and just query it when necessary?

**Review by jakobbotsch — APPROVED** [review]
LGTM, just one nit.

Looks like I lost the update of `ProfileSynthesis::m_improperLoopHeaders` all the way back in #95251.

File: src/coreclr/jit/fgprofilesynthesis.cpp (hunk 1 above)
Comment (AndyAyersMS) [reply src/coreclr/jit/fgprofilesynthesis.cpp:36]: Which field -- `m_dfsTree`?

Comment (AndyAyersMS) [reply src/coreclr/jit/fgprofilesynthesis.cpp:36]: FYI `m_improperLoopHeaders` will get more uses in a follow-on PR.

Comment (jakobbotsch) [reply src/coreclr/jit/fgprofilesynthesis.cpp:36]: `ProfileSynthesis::m_improperLoopHeaders` uses can be replaced by `m_loops->ImproperLoopHeaders()`.

**Comment by AndyAyersMS** [comment]
Kind of annoying but we can't safely check whether the profile is consistent at the point we synthesize or repair. So we run the check and remember the result, but wait for the importer to sort out the bad IL cases, and then we assert that the earlier result was what we expected (note we can't defer the entire check since the importer can mess up a consistent profile).

**Pull request merged**